src/clr-angular/forms/select/select.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {AfterContentInit, Component, ElementRef, ViewChild} from "@angular/core";
+
+import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+@Component({
+    selector: "clr-select",
+    templateUrl: "select.html",
+    providers: [IfOpenService, {provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef}],
+    host: {"[class.clr-select]": "true"}
+})
+export class ClrSelect implements AfterContentInit {
+    @ViewChild("input") input: ElementRef;
+
+    constructor(private ifOpenService: IfOpenService) {}
+
+    private registerPopoverIgnoredInput() {
+        if (this.input) {
+            this.ifOpenService.registerIgnoredElement(this.input);
+        }
+    }
+
+    onClick(event: MouseEvent): void {
```

Comment:
As a general rule, I don't like `onClick` or `onHover` as method names, it makes it very hard to read the template. The name conveys no information here. It's much better to call them by what they do, in your case it would be `toggleMenu` or something similar.

Note to self: add this to the coding guidelines.

Reply:
good point. fixed.